src/main/scala/com/databricks/spark/csv/CsvRelation.scala

```diff
@@ -123,6 +123,10 @@ case class CsvRelation protected[spark] (
           case aiob: ArrayIndexOutOfBoundsException if permissive =>
             (index until schemaFields.length).foreach(ind => rowArray(ind) = null)
             Some(Row.fromSeq(rowArray))
+          case nfe: java.lang.NumberFormatException if dropMalformed =>
```

Comment:
Just realized we are silently dropping data here. Please add a warning like this:
```
logger.warn(s"Number format error. Dropping malformed line: ${tokens.mkString(",")}")
```